lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -500,6 +501,14 @@ void OCLBackend::executeConvolution(const OCLConvolutionInst *CC) {
                                 ((M_FW_ - 1) / fw_div_M + 1) * fw_wgs1,
                                 idim.n * group};
 
+  if (fw_wgs0 == 1 && fw_wgs1 == 1) {
+    // Special case for the execution on Apple's CPU OpenCL backend, where
```

Comment:
I don't understand this comment and following code. Why is it specific to Apple? Why do we only touch dimension N, but not M?
Can you simplify loop by using division?